### PR TITLE
[`ruff`] Avoid false positives for RUF027 for typing context bindings.

### LINKED ISSUE #14000: RUF027 False positive for non-runtime context binding
This might be incredibly easy to fix depending on how this check is implemented. Currently RUF027 can be triggered by something like:

```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from datetime import date

path = "foo/{date}/"  # RUF027
```

Now, to be fair, this would also be a false positive for a runtime binding. But this seems like an easy low hanging fruit to reduce the false positive rate at least a little bit.

Although a more reliable solution long-term would be to add a setting for specifying additional methods which expect a template string.

### PR DESCRIPTION
**Pull request by Daverball**
Closes #14000 

## Summary

For typing context bindings we know that they won't be available at runtime. We shouldn't recommend a fix, that will result in name errors at runtime.

## Test Plan

`cargo nextest run`

### CODE REVIEW

File: crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs [hunk 1]
```diff
@@ -209,7 +209,15 @@ fn should_be_fstring(
...
-                    .lookup_symbol(id)
+                    // the parsed expression nodes have incorrect ranges
+                    // so we create a dummy node which has a range that's
+                    // close enough to being correct in order for
+                    // `simulate_runtime_load` to return the correct result.
+                    .simulate_runtime_load(&ast::ExprName {
+                        id: id.clone(),
+                        range: literal.range(),
+                        ctx: ast::ExprContext::Load,
+                    })
```
Comment (MichaReiser) [inline crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs:220]: I think I have a slight preference to adding an extra method here that takes a string and range over creating a fake AST
```suggestion
                    .simulate_runtime_load_str(&id, literal.range())
```

**Review by MichaReiser — APPROVED** [review]
Nice

File: crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs (hunk 1 above)
Comment (Daverball) [reply crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs:220]: I considered that as well. My most favorite solution would be to extend `Parser` with an optional static offset, so the parsed nodes get created with the correct offset.

**Review by MichaReiser — APPROVED** [review]
Thanks

**Pull request merged**